src/main/java/seedu/realodex/model/person/predicates/BirthdayIsInMonthPredicate.java

```diff
@@ -0,0 +1,76 @@
+package seedu.realodex.model.person.predicates;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import seedu.realodex.commons.util.ToStringBuilder;
+import seedu.realodex.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Birthday} is in the Month given.
+ */
+public class BirthdayIsInMonthPredicate implements Predicate<Person> {
+    private final SimpleDateFormat monthFormat = new SimpleDateFormat("MMMM");
+    private final Optional<Calendar> month;
+
+    /**
+     * Creates object {@code BirthdayIsInMonthPredicate}
+     * @param monthName month to be tested
+     */
+    public BirthdayIsInMonthPredicate(String monthName) {
+        Optional<Calendar> tempMonth;
+        Date monthDate;
+        try {
+            monthDate = monthFormat.parse(monthName);
+            tempMonth = Optional.of(Calendar.getInstance());
+            tempMonth.get().setTime(monthDate);
+        } catch (java.text.ParseException e) {
+            // will not reach here because Parser checks for valid month
+            assert false;
+            tempMonth = Optional.empty();
+        }
+
+        this.month = tempMonth;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        Calendar personCalendar = Calendar.getInstance();
+        if (person.getBirthday().birthday.isEmpty()) {
+            return false;
+        } else {
+            assert(person.getBirthday().birthday.isPresent());
+            personCalendar.setTime(person.getBirthday().birthday.get());
+            return personCalendar.get(Calendar.MONTH) == month.get().get(Calendar.MONTH);
+        }
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof BirthdayIsInMonthPredicate)) {
+            return false;
+        }
+
+        BirthdayIsInMonthPredicate otherBirthdayIsInMonthPredicate = (BirthdayIsInMonthPredicate) other;
+        return month.equals(otherBirthdayIsInMonthPredicate.month);
+    }
+
+    @Override
+    public String toString() {
+        if (this.month.isPresent()) {
+            return new ToStringBuilder(this).add("month", monthFormat.format(month.get().getTime())).toString();
+        }
+        // will not reach here because Parser checks for valid month
+        assert false;
+        return new ToStringBuilder(this).add("month", "No month specified").toString();
```

Comment:
```suggestion
```

No need to include this part, returning on top is sufficient as the program will never reach here anyway. Assertions also cannot be used for control flow as shown here. Just return the tostringbuilder. In the first place, this object would not even be instantiated because of your assertion in the constructor 😄

Reply:
i got to return something otherwise it won't compile cus toString must return string